azure_storage_client.go

```diff
@@ -276,16 +278,33 @@ func (util *snowflakeAzureClient) nativeDownloadFile(
 	if meta.mockAzureClient != nil {
 		blobClient = meta.mockAzureClient
 	}
-	f, err := os.OpenFile(fullDstFileName, os.O_CREATE|os.O_WRONLY, readWriteFileMode)
-	if err != nil {
-		return err
-	}
-	defer f.Close()
-	_, err = blobClient.DownloadFile(
-		context.Background(), f, &azblob.DownloadFileOptions{
-			Concurrency: uint16(maxConcurrency)})
-	if err != nil {
-		return err
+	if meta.options.getFileToStream {
+		blobDownloadResponse, err := blobClient.DownloadStream(context.Background(), &azblob.DownloadStreamOptions{})
+		if err != nil {
+			return err
+		}
+		retryReader := blobDownloadResponse.NewRetryReader(context.TODO(), &azblob.RetryReaderOptions{})
+		defer retryReader.Close()
+		buf := bytes.NewBuffer(meta.dstStream)
+		_, err = buf.ReadFrom(retryReader)
+		if err != nil {
+			return err
+		}
+		if buf != nil {
+			meta.dstStream = buf.Bytes()
```

Comment:
Yes, it will be the same as WithFileStream(). The `Downloader.Download()` API doesn't seem to have an option to load by chunks. Optimization can be done as part of https://github.com/snowflakedb/snowflake-sdks-drivers-issues-teamwork/issues/854 in the future.